x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -441,15 +441,19 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     return !!(scalingType === SCALING_TYPES.TOP_HITS && topHitsSplitField);
   }
 
-  async supportsFeatureEditing(): Promise<boolean> {
+  async getSourceIndexList(): Promise<string[]> {
     await this.getIndexPattern();
     if (!(this.indexPattern && this.indexPattern.title)) {
-      return false;
-    }
-    const { matchingIndexes } = await getMatchingIndexes(this.indexPattern.title);
-    if (!matchingIndexes) {
-      return false;
+      return [];
     }
+    const {
+      payload: { matchingIndexes },
+    } = await getMatchingIndexes(this.indexPattern.title);
+    return matchingIndexes;
```

Comment:
You need to handle the case where matchingIndexes is undefined and return empty array in that case. This can be handled here or in getMatchingIndexes. Maybe it would be better if handled in getMatchingIndexes and getMatchingIndexes could just return an string[] and better encapsulate the API response.

Reply:
To be consistent with how we're handling other endpoints, I've made it conditional on whether the call was successful or not. Successful calls to the endpoint always return an array. Unsuccessful calls don't which we now handle here by detecting and returning an empty array from the function.